lib/db_utils.pm

```diff
@@ -69,24 +75,35 @@ sub influxdb_push_data {
 =head2 influxdb_read_data
 
 Builds an Influx DB query and read data from specified database
-with C<url_base> and C<db> for the Influx DB name. C<query> contains
-SELECT query for given DB.
+with C<url_base>, C<db> and C<org> for the Influx DB name.
+C<token> contains the access token.
+C<query> contains SELECT query for given DB.
 
 returns json with results of SELECT query.
 
 =cut
 
 sub influxdb_read_data {
-    my ($url_base, $db, $query) = @_;
+    my ($url_base, $db, $org, $token, $query) = @_;
+
     my $ua = Mojo::UserAgent->new();
     $ua->max_redirects(5);
-    my $mojo_url = Mojo::URL->new($url_base . '/query');
-    $mojo_url->query(db => $db, q => $query);
-    my $res = $ua->get($mojo_url)->res;
-    unless ($res && $res->json) {
+
+    my $mojo_url = Mojo::URL->new("$url_base/query");
+    $mojo_url->query(orgID => $org, db => $db, q => $query);
+    my $res = $ua->post($mojo_url => {Authorization => "Token $token", Accept => "application/json", 'Content-type' => "application/json"})->res;
+    if ($res->code != 200) {
         die sprintf("Failed to get data from InfluxDB. \n Response code : %s \n Message: %s \n", $res->code, $res->message);
+    } else {
+        my $json = $res->json;
+        my @results = $json->{results};
+        # Check if we got an error object as reply
+        my $result = $results[0][0];
+        if (defined($result->{error})) {
```

Comment:
is it possible to get HTTP=200 but error inside ?

Reply:
Yes that happens. Don't ask me why, but it's rather common.

Reply:
ok I won't ask *you* why :)